apps/src/lib/metrics/MetricsReporter.ts

```diff
@@ -34,27 +34,64 @@ class MetricsReporter {
       parseInt(localStorage.getItem(LOCAL_STORAGE_KEY_NAME) || '0') || 0;
   }
 
+  /**
+   * Publish an information log message. Can be a string or a structured object
+   */
   logInfo(message: string | object) {
     this.log('INFO', message);
     if (isDevelopmentEnvironment()) {
       console.log('[MetricsReporter] ' + JSON.stringify(message));
     }
   }
 
+  /**
+   * Publish a warning log message. Can be a string or a structured object
+   */
   logWarning(message: string | object) {
     this.log('WARNING', message);
     if (isDevelopmentEnvironment()) {
       console.warn('[MetricsReporter] ' + JSON.stringify(message));
     }
   }
 
+  /**
+   * Publish an error log message. Can be a string or a structured object
+   */
   logError(message: string | object) {
     this.log('SEVERE', message);
     if (isDevelopmentEnvironment()) {
       console.error('[MetricsReporter] ' + JSON.stringify(message));
     }
   }
 
+  /**
+   * Increment a counter metric.
+   */
+  incrementCounter(name: string, dimensions: MetricDimension[] = []) {
+    this.publishMetric(name, 1, 'Count', dimensions);
+  }
+
+  /**
+   * Publish a metric.
+   */
+  publishMetric(
+    name: string,
+    value: number,
+    unit: MetricUnit,
+    dimensions: MetricDimension[] = []
+  ) {
+    const metric = {
+      name,
+      value,
+      unit,
+      dimensions: dimensions.concat(this.getDeviceDimensions()),
+    };
+    this.sendMetric(metric);
+    if (isDevelopmentEnvironment()) {
```

Comment:
out of curiosity, where do logs go on the development environment? Could `sendMetric` just write to `console.info`?